pkgs/development/interpreters/php/libxml212.patch

```diff
@@ -0,0 +1,55 @@
+diff --git a/ext/libxml/libxml.c b/ext/libxml/libxml.c
+index 3959b362a0..6cdfbd397f 100644
+--- a/ext/libxml/libxml.c
++++ b/ext/libxml/libxml.c
+@@ -483,7 +483,11 @@ static void _php_libxml_free_error(void *ptr)
+ 	xmlResetError((xmlErrorPtr) ptr);
+ }
+ 
+-static void _php_list_set_error_structure(xmlErrorPtr error, const char *msg)
++#if LIBXML_VERSION >= 21200
++static void _php_list_set_error_structure(const xmlError *error, const char *msg)
++#else
++static void _php_list_set_error_structure(xmlError *error, const char *msg)
++#endif
+ {
+ 	xmlError error_copy;
+ 	int ret;
+@@ -736,7 +740,11 @@ PHP_LIBXML_API void php_libxml_ctx_warning(void *ctx, const char *msg, ...)
+ 	va_end(args);
+ }
+ 
++#if LIBXML_VERSION >= 21200
++PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, const xmlError *error)
++#else
+ PHP_LIBXML_API void php_libxml_structured_error_handler(void *userData, xmlErrorPtr error)
++#endif
+ {
+ 	_php_list_set_error_structure(error, NULL);
+ 
+@@ -1009,11 +1017,9 @@ PHP_FUNCTION(libxml_use_internal_errors)
+ /* {{{ Retrieve last error from libxml */
+ PHP_FUNCTION(libxml_get_last_error)
+ {
+-	xmlErrorPtr error;
+-
+ 	ZEND_PARSE_PARAMETERS_NONE();
+ 
+-	error = xmlGetLastError();
++	const xmlError *error = xmlGetLastError();
+ 
+ 	if (error) {
+ 		object_init_ex(return_value, libxmlerror_class_entry);
+diff --git a/ext/soap/php_sdl.c b/ext/soap/php_sdl.c
+index 651eab23b7..7a7ce304d6 100644
+--- a/ext/soap/php_sdl.c
++++ b/ext/soap/php_sdl.c
```

Comment:
Would not this patch need to be applied to soap in `php-packages.nix` as well?

Reply:
If it's applied on PHP, I don't think (_I might be wrong_) it's needed to apply it on the `soap` and `libxml` extensions.

Reply:
Unlike `libxml` extension, we build soap separately from `php.unwrapped`. You can see in https://logs.ofborg.org/?key=nixos/nixpkgs.283175&attempt_id=f8dd8424-de95-4d97-b2d2-2c752497310b that this patch is not applied when building `soap`.

However, it looks like it just fixes a warning so it is not currently strictly necessary. We should either apply it to soap extension too, or drop the soap part.